tools/scripts/run_protocol_tests.py

```diff
@@ -0,0 +1,149 @@
+﻿#!/usr/bin/env python3
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0.
+
+"""
+A script to run protocol tests for AWS SDK for C++:
+Provided --build_dir, it runs each built protocol test found.
+For each test executed, a mock http server is launched in parallel.
+"""
+import argparse
+import os
+import shutil
+import signal
+import subprocess
+import time
+
+SCRIPTS_DIR = os.path.dirname(__file__)
+PROTO_TEST_MOCK_HANDLER = f"{SCRIPTS_DIR}/protocol_tests_mock_server/protocol_tests_mock_server.py"
+
+
+def parse_arguments() -> dict:
+    """Parse arguments and/or print help
+
+    :return: dict with parsed arguments
+    """
+    parser = argparse.ArgumentParser(description="Protocol tests driver for AWS-SDK-CPP")
+    parser.add_argument("--debug", action="store_true", help="Print additional information")
+    parser.add_argument("--build_dir", action="store", required=True, help="Path to a directory with built SDK")
+    args = vars(parser.parse_args())
+    arg_map = {"debug": args.get("debug", False)}
+    arg_map["build_dir"] = args.get("build_dir")
+    return arg_map
+
+
+class MockHttpServerHandle:
+    """A RAII-style helper wrapper around mini mock http server.
+    Launches a server on construction and kills it on destruction
+    """
+
+    def __init__(self):
+        try:
+            python_cmd = [shutil.which("python3"), PROTO_TEST_MOCK_HANDLER]
```

Comment:
sys.executable if you want to make sure its the same python?

Reply:
No need to be sure it is the same interpreter
- test driver
- mock server
- test app
All can be seen as an independent apps.